guides/v2.3/graphql/develop/resolvers.md

```diff
@@ -42,7 +15,185 @@ $info | [`Magento\Framework\GraphQl\Schema\Type\ResolveInfo`]({{ site.mage2blobu
+    {
+        return [ProductLinksRetriever::class, 'getRelatedProducts'];
+    }
+
+    public function convertToServiceArgument(ResolveRequestInterface $request)
+    {
+        return new RootProductCriteria($request->getValue()['model']->getId());
+    }
+
+    public function convertFromServiceResult($result, ResolveRequestInterface $request)
+    {
+        return $result->getLinkedProducts();
+    }
+}
+```
+
+The `getServiceContract()` method points to the service contract to be used.
+
+The `convertToServiceArgument()` method converts GraphQL requests to a criteria item to be passed in a list as the argument to the contract. Remember that batch service contract methods must accept a single argument: a list (array) of criteria objects.
+
+The `convertFromServiceResult()` method converts one of the result items into a GraphQL response (a [`\Magento\Framework\GraphQl\Query\Resolver\Value`]({{ site.mage2bloburl }}/{{ page.guide_version }}/lib/internal/Magento/Framework/GraphQl/Query/Resolver/Value.php) instance or an array). Remember that batch service contracts must return result items in the same order as were the criteria items passed as the method's list argument. For example, if the first root product had the ID #555, then the first result item will contain the list of related products to product #555.
+
+The batch service contract used in the example would look something like this:
+
+```php
+class ProductLinksRetriever
+{
+    ...
+
+    /**
+     * @param RootProductCriteria[] $criteriaList
+     * @return RelatedProductsFound[]
+     */
+    public function getRelatedProducts(array $criteriaList): array
+    {
+        ....
+    }
+}
+
+class RootProductCriteria
+{
+    ....
+
+    public function __construct(int $rootProductId)
+    {
+        $this->productId = $rootProductId;
+    }
+
+    public function getRootProductId(): int
+    {
+        return $this->productId;
+    }
+}
+
+class RelatedProductsFound
+{
+    ....
+
+    public function getLinkedProducts(): array
+    {
+        ....
+    }
+
+    public function getRootProductId(): int
+    {
+        ....
+    }
+}
+```
+
+A real example can be found at [\Magento\CatalogGraphQl\Model\Resolver\Product\BatchProductLinks]({{ site.mage2bloburl }}/{{ page.guide_version }}/app/code/Magento/CatalogGraphQl/Model/Resolver/Product/BatchProductLinks.php)
+
+### ResolverInterface
+
+This resolver resolves one branch or leaf at a time. The interface returns [`\Magento\Framework\GraphQl\Query\Resolver\Value`]({{ site.mage2bloburl }}/{{ page.guide_version }}/lib/internal/Magento/Framework/GraphQl/Query/Resolver/Value.php) or any type of data (mixed). This value takes in a callable function to its constructor that will be invoked at the latest possible time for the resolver to acquire its data. As a result, a list of items being resolved can be retrieved all at once by establishing a buffer that contains all relevant parent data to filter and fetch for the children list data.
```

Comment:
```suggestion
This interface resolves one branch or leaf at a time. It returns [`\Magento\Framework\GraphQl\Query\Resolver\Value`]({{ site.mage2bloburl }}/{{ page.guide_version }}/lib/internal/Magento/Framework/GraphQl/Query/Resolver/Value.php) or any type of data (mixed). This value takes in a callable function to its constructor that will be invoked at the latest possible time for the resolver to acquire its data. As a result, a list of items being resolved can be retrieved all at once by establishing a buffer that contains all relevant parent data to filter and fetch for the children list data.
```

Reply:
this bit already was in the document, I just moved it